Spoony-iOS/Spoony-iOS/Source/Feature/Home/Components/BottomSheet/CustomBottomSheet.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  CustomBottomSheet.swift
+//  SpoonMe
+//
+//  Created by 이지훈 on 1/12/25.
+//
+
+import SwiftUI
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
+}
+
+struct RoundedCorner: Shape {
+    var radius: CGFloat = .infinity
+    var corners: UIRectCorner = .allCorners
+
+    func path(in rect: CGRect) -> Path {
+        let path = UIBezierPath(roundedRect: rect,
+                               byRoundingCorners: corners,
+                               cornerRadii: CGSize(width: radius, height: radius))
+        return Path(path.cgPath)
+    }
+}
+
+struct CustomBottomSheet<Content: View>: View {
+    let style: BottomSheetStyle
+    let content: Content
+    @Binding var isPresented: Bool
+    @State private var currentHeight: CGFloat
+    @State private var dragOffset: CGFloat = 0
+    
+    @GestureState private var isDragging: Bool = false
+    @State private var velocity: CGFloat = 0
+    
+    init(style: BottomSheetStyle, isPresented: Binding<Bool>, @ViewBuilder content: () -> Content) {
+        self.style = style
+        self._isPresented = isPresented
+        self.content = content()
+        self._currentHeight = State(initialValue: style.height)
+    }
+    
+    private var maxHeight: CGFloat {
+        BottomSheetStyle.full.height
+    }
+    
+    private var minHeight: CGFloat {
+        BottomSheetStyle.minimal.height
+    }
+    
+    private func handleDragGesture(value: DragGesture.Value) {
+        let verticalVelocity = value.predictedEndLocation.y - value.location.y
+        
+        if style == .button {
+            if value.translation.height > 30 || verticalVelocity > 100 {
+                withAnimation(.spring(response: 0.2, dampingFraction: 0.8, blendDuration: 0)) {
+                    dragOffset = 0
+                    isPresented = false
+                }
+            } else {
+                withAnimation(.spring(response: 0.2, dampingFraction: 0.8, blendDuration: 0)) {
+                    dragOffset = 0
+                }
+            }
+            return
+        }
```

Comment:
dragOffset = 0 로직이 중복되어 보입니다!
애니메이션 효과를 둘 다 넣고 싶은 의도 라면, 아예 withAnimation을 빼가지고 아래처럼 할 수 있을 거 같아요!

```suggestion
if style == .button {
    withAnimation(.spring(response: 0.2, dampingFraction: 0.8, blendDuration: 0)) {
        dragOffset = 0
        if value.translation.height > 30 || verticalVelocity > 100 {
            isPresented = false
        }
    }
    return
}
```